app/code/core/Mage/Core/Model/Encryption.php

```diff
@@ -126,6 +126,9 @@ public function hash($data, $version = self::HASH_VERSION_MD5)
      */
     public function validateHash($password, $hash)
     {
+        if (strlen($password) > Mage_Customer_Model_Customer::MAXIMUM_PASSWORD_LENGTH)
```

Comment:
I think the constant shouldn't be "coupled" with the Mage_Customer module, maybe it could be in this same Model_Encryption file?